### PR TITLE
feat: add Chinese docs translation

### PR DESCRIPTION
**Pull request by DDDDD12138**
### Description
Add Chinese documentation, detailed comparison with the original English can be found in this [commit](https://github.com/vuejs/test-utils/commit/3f3f8b7041f6581a9905da99d5620a3ce58a9fc1).

### Preview
image
image
image

### CODE REVIEW

**Review by Jinjiang — CHANGES_REQUESTED** [review]
Well, after reviewing some docs, I've found quite a lot of formatting issues. I would recommend you to refer to this [Chinese translation guide](https://github.com/vuejs-translations/docs-zh-cn/wiki/%E7%BF%BB%E8%AF%91%E9%A1%BB%E7%9F%A5) for more details. And in practice, you can feel free to run [zhlint](https://www.npmjs.com/package/zhlint), the tool we use for Vue Chinese docs, on your translation files. Usually for most of the issues, this tool can help you to figure them out and fix them. We can discuss further if you have any questions in any particular cases.

After solving the formatting issues, we can merge it ahead (probably with a "in review" marks on the pages somewhere) and then review the docs piece by piece as the second step.

Thanks.

**Comment by DDDDD12138** [comment]
ok, I'll deal with it later.

File: docs/zh/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Vue Test Utils
```
[the remark this answers is not in the document]
Comment (DDDDD12138) [reply docs/zh/README.md:1]: Does Chinese not require this file? I referred to fr, and it added this file (although I think I forgot to translate it).

File: docs/zh/guide/essentials/a-crash-course.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+test('renders a todo', () => {
+  const wrapper = mount(TodoApp)
+
+  const todo = wrapper.get('[data-test="todo"]')
+
+  expect(todo.text()).toBe('Learn Vue.js 3')
+})
+```
+
+我们首先导入 `mount` —— 这是在 VTU 中渲染组件的主要方式。使用 `test` 函数声明一个测试，并给出简短的描述。`test` 和 `expect` 函数在大多数测试运行器中是全局可用的 (此示例使用 [Jest](https://jestjs.io/))。如果 `test` 和 `expect` 看起来令人困惑，Jest 文档中有一个[更简单的示例](https://jestjs.io/docs/getting-started)，可以帮助你理解它们的用法和工作原理。
```
Comment (Jinjiang) [inline docs/zh/guide/essentials/a-crash-course.md:55]: 中文破折号两边不需要空格 (因为是全角符号)，下同

File: docs/zh/guide/advanced/async-suspense.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+你可能注意到在指南的某些部分，在调用 `wrapper` 的一些方法时使用了 `await`，例如 `trigger` 和 `setValue`。这是什么意思呢？
+
+你可能知道 [Vue 是以响应式的方式更新的](https://v3.vuejs.org/guide/change-detection.html#async-update-queue)：当你更改一个值时，DOM 会自动更新以反映最新的值。Vue 的这些更新是异步进行的。与此相对，像 Jest 这样的测试运行器是 _同步_ 的。这可能会导致测试中出现一些意外的结果。
+
+让我们看看一些策略，以确保在运行测试时 Vue 按预期更新 DOM。
+
+## 简单示例 - 使用 `trigger` 更新
+
+让我们重用来自 [事件处理](../essentials/event-handling) 的 `<Counter>` 组件，唯一的变化是我们现在在 `template` 中渲染 `count`。
```
Comment (Jinjiang) [inline docs/zh/guide/advanced/async-suspense.md:11]:
```suggestion
让我们重用来自[事件处理](../essentials/event-handling)的 `<Counter>` 组件，唯一的变化是我们现在在 `template` 中渲染 `count`。
```

md 标记内外都是中文或其它全角字符的情况下不需要空格，下同

File: docs/zh/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jinjiang) [reply docs/zh/README.md]: Not necessary unless you have anything extra to write about as a translator.

File: docs/.vitepress/theme/index.mts [hunk 4]
```diff
@@ -5,6 +5,7 @@ import status from '../translation-status.json'
 import './custom.css'
 const i18nLabels = {
   fr: 'La traduction est synchronisée avec les docs du ${date} dont le hash du commit est <code>${hash}</code>.',
+  zh: '翻译与${date}的文档同步，其 commit 的哈希值为 <code>${hash}</code>。'
```
Comment (Jinjiang) [inline docs/.vitepress/theme/index.mts:8]:
```suggestion
  zh: '该翻译已同步到了 ${date} 的版本，其对应的 commit hash 是 <code>${hash}</code>。'
```

I just synced the translation in other official libs like pinia and router. And also the date needs spaces around. 🙏

File: docs/zh/migration/index.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+        }
+      }
+    }
+  }
+  ```
+- 通过 `@vue/test-utils` [`mountingOptions.global.config`](https://test-utils.vuejs.org/api/#global) 全局或逐个测试基础。
+
+## 与 v1 的比较
+
+这是一个针对来自 VTU 1 的用户的表格，比较 API。
```
Comment (veaba) [inline docs/zh/migration/index.md:349]:
```suggestion
这是一个针对来自 VTU 1 用户的表格，其比较了两者的 API。
```

File: docs/zh/guide/advanced/reusability-composition.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    axios.get(`users/${id}`).then((response) => (user.value = response.data))
+  }
+
+  onMounted(() => fetchUser(userId))
+
+  return { user }
+}
+```
+
+要测试这个组合式函数，你可以在测试中创建一个简单的 `TestComponent`。`TestComponent` 应该以与真实组件相同的方式使用组合式函数。
```
Comment (veaba) [inline docs/zh/guide/advanced/reusability-composition.md:54]:
```suggestion
要测试这个组合式函数，你可以在测试中创建一个简单的 `TestComponent`。`TestComponent` 应该与真实组件一样的方式使用组合式函数。
```
优化表达

File: docs/zh/guide/advanced/reusability-composition.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+  await flushPromises()
+
+  expect(wrapper.vm.user).toEqual({ id: 1, name: 'User' })
+})
+```
+
+## Provide (提供) / Inject (注入)
+
+Vue 提供了一种通过 `provide` 和 `inject` 将 props 传递给所有子组件的方法。测试这种行为的最佳方式是测试整个树 (父组件 + 子组件)。但有时这并不可能，因为树结构过于复杂，或者你只想测试单个组合式函数。
```
Comment (veaba) [inline docs/zh/guide/advanced/reusability-composition.md:93]:
```suggestion
Vue 提供了一种通过 `provide` 和 `inject` 将 props 传递给所有子组件的方法。测试这种行为的最佳方式是测试整个树 (父组件+子组件)。但有时这并不可能，因为树结构过于复杂，或者你只想测试单个组合式函数。
```
中文之间的空格使用

**Review by veaba — COMMENTED** [review]
文件太大没法加 sug，斜体使用

```md
另外补充：vue-router.md L289 部分。
```suggestion
然而，在这种情况下，我们没有可以等待的 *hasNavigated* 钩子。一种替代方法是使用从 Vue Test Utils 导出的 `flushPromises` 函数：
```

添加太大 500 了。
```

File: docs/zh/guide/advanced/reusability-composition.md (hunk 7 above)
Comment (Jinjiang) [reply docs/zh/guide/advanced/reusability-composition.md:93]: 都说得通吧，这里我个人倾向是有空格的，因为加号相当于是半角符号

Comment (DDDDD12138) [reply docs/zh/guide/advanced/reusability-composition.md:93]: 那就都保持这样吧？带上空格也显得不会那么挤

**Comment by DDDDD12138** [comment]
> 添加太大 500 了。

这个指的是啥？我这边好像没显示出来🤔

**Comment by veaba** [comment]
> > 添加太大 500 了。
> 
> 这个指的是啥？我这边好像没显示出来🤔

那个 vue-router.md 文件太大，没办法对指定位置添加 Suggetion，Github 报 500了。

改动部位：`_hasNavigated_` 改为 `*hasNavigated*` 就可以了。


@Jinjiang  我这边 review 结束了，再交叉检查下应该可以了。

**Review by Jinjiang — APPROVED** [review]

**Comment by Jinjiang** [comment]
image

I've added these in-review labels and created this epic issue #2561 for further review.

For this PR I will merge soon. Thanks all.

**Pull request merged**